### PR TITLE
design(game_env): Design

### LINKED ISSUE #70: Design in Game Env API

### PR DESCRIPTION
**Pull request by EdwardHaoranLee**
Come up the structure for the ML interface for the game.

### Motivation and context
#70 

### Describe your changes
- Added file learn/src/data/woodoku_game_env.py and corresponding docstring and method signature. 

### Tests performed
N/A

### Before submitting this PR, please make sure:
 - [x] Your code builds cleanly without any errors

### CODE REVIEW

File: learn/src/data/woodoku_game_env.py [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+
+class Observation:
+    """
+    The class representing the observation.
+
+    The observation space is a 1D tensor that has the following length. One board representation and three shapes to
+    choose from.
+
+    The board representation part will be integer 0 or 1 indicating whether that block is occupied. Each shape will also
+    be 0 or 1 and will reflect the actual shape, and will be totally empty if is not available to choose.
```
Comment (Xiaorui-Huang) [inline learn/src/data/woodoku_game_env.py]: States missing, streaks and combo

File: learn/src/data/woodoku_game_env.py [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+    _availability: List[bool]
+
+    def __init__(self) -> None:
+        self._woodoku_board = WoodokuBoard()
+        self._all_shapes = game.read_shapes_from_file(game.CONFIG_FILE)
+        self._shapes = game.random_shapes(self._all_shapes, game.NUM_SHAPES)
+        self._availability = [True] * game.NUM_SHAPES
+
+    def reset(self) -> None:
+        """
```
Comment (Xiaorui-Huang) [inline learn/src/data/woodoku_game_env.py:56]: should reset just call __init__?

File: learn/src/data/woodoku_game_env.py [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+        Update self.data based on the woodoku board and the shapes.
+
+        Args:
+            board: The Woodoku Board.
+            shapes: The shapes to choose.
+        """
+        pass
+
+
+class Action:
```
Comment (Xiaorui-Huang) [inline learn/src/data/woodoku_game_env.py]: Are random actions or even actions part of the game env? Could just be a function in the Learn package

File: learn/src/data/woodoku_game_env.py (hunk 2 above)
Comment (EdwardHaoranLee) [reply learn/src/data/woodoku_game_env.py:56]: Let's see.

**Review by Xiaorui-Huang — APPROVED** [review]

File: learn/src/data/woodoku_game_env.py (hunk 1 above)
Comment (EdwardHaoranLee) [reply learn/src/data/woodoku_game_env.py]: Fixed in [90cc4e8](https://github.com/EdwardHaoranLee/WoodokuLearn/pull/71/commits/90cc4e8f80a58739665d5e00e74f2d249d83b94e)

File: learn/src/data/woodoku_game_env.py (hunk 3 above)
Comment (EdwardHaoranLee) [reply learn/src/data/woodoku_game_env.py]: Fixed in [90cc4e8](https://github.com/EdwardHaoranLee/WoodokuLearn/pull/71/commits/90cc4e8f80a58739665d5e00e74f2d249d83b94e)

**Pull request merged**